x/erc20/keeper/ibc_callbacks.go

```diff
@@ -82,36 +80,51 @@ func (k Keeper) OnRecvPacket(
 	)
 
 	// check if the coin is a native staking token
-	bondDenom := k.stakingKeeper.BondDenom(ctx)
-	if coin.Denom == bondDenom {
+	if coin.Denom == evmParams.EvmDenom {
 		// no-op, received coin is the staking denomination
 		return ack
 	}
 
 	pairID := k.GetTokenPairID(ctx, coin.Denom)
-	if len(pairID) == 0 {
-		// short-circuit: if the denom is not registered, conversion will fail
-		// so we can continue with the rest of the stack
-		return ack
-	}
+	pair, found := k.GetTokenPair(ctx, pairID)
+	switch {
+	// Case 1. token pair is not registered and is a single hop IBC Coin
+	case !found && ibc.IsSingleHop(coin.Denom):
+		contractAddr, err := utils.GetIBCDenomAddress(coin.Denom)
+		if err != nil {
+			return channeltypes.NewErrorAcknowledgement(err)
+		}
+
+		found := evmParams.IsPrecompileRegistered(contractAddr.String())
```

Comment:
Is it possible to not have the TokenPair but have the precompile registered?

Reply:
Just so the same question as a comment

Reply:
As per this statement no:

https://github.com/evmos/evmos/blob/03395c1044079ae65406bd3a5c157aea273cd9c3/x/erc20/keeper/ibc_callbacks.go#L110-L110